Prox/Prox/Data/PlacesController.swift

```diff
@@ -187,10 +180,20 @@ class PlacesProvider {
         delegate?.placesProvider(self, didUpdatePlaces: displayedPlaces)
     }
 
+    /// Returns a copy of all the places. Callers should not be locked.
+    func getAllPlacesCopy() -> [Place] {
+        return getCopy(ofPlaces: allPlaces)
+    }
+
+    /// Returns a copy of the displayed places. Callers should not be locked.
     func getDisplayedPlacesCopy() -> [Place] {
+        return getCopy(ofPlaces: displayedPlaces)
+    }
+
+    private func getCopy(ofPlaces places: [Place]) -> [Place] {
         var placesCopy: [Place] = []
         placesLock.withReadLock {
-            placesCopy = Array(self.displayedPlaces)
+            placesCopy = Array(places)
```

Comment:
You don't need the `Array()` initializer since Swift arrays are already copy-on-write (I'd also remove `Copy` from the names since that's standard Swift behavior).